### PR TITLE
fix #1745 Deprecate compose/transform and replace with aliases

### LINKED ISSUE #1745: Deprecate Flux/Mono#compose
`transform` and `compose` difference is never obvious when looking at the names only.
`compose` can be achieved with `transform`+`defer`, so if we were to rationalize down to a single method, `transform` would be the 1s candidate.

### PR DESCRIPTION
**Pull request by simonbasle**
This makes the relationship between the two clearer, while also better
showing their difference in behavior at a glance:
 - composeNow immediately applies the Function
 - composeDefer mixes defer and composeNow behaviors

### CODE REVIEW

**Comment by simonbasle** [comment]
note there is also `transform` in `ParallelFlux`, so I aligned it with Flux and Mono and changed to `composeNow` + deprecation. There is no `compose`, but a `composeGroup` (with a slightly different behavior, leaving as is).

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 1]
```diff
@@ -3397,20 +3397,66 @@ public final T blockLast(Duration timeout) {
...
-	 * <img class="marble" src="doc-files/marbles/composeForFlux.svg" alt="">
+	 * <img class="marble" src="doc-files/marbles/composeLazilyForFlux.svg" alt="">
 	 *
 	 * @param transformer the {@link Function} to lazily map this {@link Flux} into a target {@link Publisher}
 	 * instance for each new subscriber
 	 * @param <V> the item type in the returned {@link Publisher}
 	 *
 	 * @return a new {@link Flux}
-	 * @see #transform  transform() for immmediate transformation of {@link Flux}
+	 * @see #composeNow  transform() for immmediate transformation of {@link Flux}
```
Comment (bsideup) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java]:
```suggestion
	 * @see #composeNow  composeNow() for immmediate transformation of {@link Flux}
```

File: reactor-core/src/main/java/reactor/core/publisher/Flux.java [hunk 2]
```diff
@@ -8724,9 +8770,11 @@ public final void subscribe(Subscriber<? super T> actual) {
 	 * @return a new {@link Flux}
 	 * @see #compose(Function) for deferred composition of {@link Flux} for each {@link Subscriber}
```
Comment (bsideup) [inline reactor-core/src/main/java/reactor/core/publisher/Flux.java]:
```suggestion
	 * @see #composeLazily(Function) for deferred composition of {@link Flux} for each {@link Subscriber}
```

File: reactor-core/src/main/java/reactor/core/publisher/Mono.java [hunk 3]
```diff
@@ -4328,9 +4378,11 @@ public final void subscribe(Subscriber<? super T> actual) {
 	 * @return a new {@link Mono}
 	 * @see #compose(Function) compose(Function) for deferred composition of {@link Mono} for each {@link Subscriber}
```
Comment (bsideup) [inline reactor-core/src/main/java/reactor/core/publisher/Mono.java]:
```suggestion
	 * @see #composeLazily(Function) composeLazily(Function) for deferred composition of {@link Mono} for each {@link Subscriber}
```

**Pull request merged**